python/src/robyn/modeling/clusters/model_clusterer.py

```diff
@@ -0,0 +1,101 @@
+# pyre-strict
+
+from typing import List, Dict, Any, Optional
+import pandas as pd
+import numpy as np
+from dataclasses import dataclass
+from sklearn.cluster import KMeans
+from sklearn.decomposition import PCA
+from sklearn.manifold import TSNE
+
+
+@dataclass
+class ClusteringResult:
+    data: pd.DataFrame
+    df_cluster_ci: pd.DataFrame
+    n_clusters: int
+    boot_n: int
+    sim_n: int
+    errors_weights: List[float]
+    clusters_means: pd.DataFrame
+    models: pd.DataFrame
+    wss_data: pd.DataFrame
+    pca_data: pd.DataFrame
+    tsne_data: pd.DataFrame
+
+
+class ModelClusterer:
+    def __init__(self, input_data: pd.DataFrame, dep_var_type: str, all_media: List[str]):
+        self.input_data = input_data
+        self.dep_var_type = dep_var_type
+        self.all_media = all_media
+
+    def cluster_models(
+        self,
+        cluster_by: str = "hyperparameters",
+        k: str = "auto",
+        wss_var: float = 0.06,
+        max_clusters: int = 10,
+        limit: int = 1,
+        weights: List[float] = [1, 1, 1],
+        dim_red: str = "PCA",
```

Comment:
What are other possible values for dim_red? Let's define enum if necessary so that we can avoid use passing unsupported value and also, it can avoid checking if value is out of all supported/valid strings.